### PR TITLE
Guzzle6 (using PSR7)

### PR DESCRIPTION
**Pull request by rubensayshi**
depends on https://github.com/99designs/http-signatures-php/pull/22 (PSR-7 support)

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -19,10 +19,16 @@
             "": "src/"
         }
     },
+    "repositories": [
+        {
+            "type": "vcs",
+            "url": "https://github.com/rubensayshi/http-signatures-php"
```
Comment (rubensayshi) [inline composer.json]: this needs to be changed once https://github.com/99designs/http-signatures-php/pull/22 is merged (and before this PR can be merged)

**Comment by rubensayshi** [comment]
after the PSR-7 PR this repo is actually becoming VERY small, maybe we can just move the class to `http-signatures-php`

the only reason I can think of for not doing that is because we've experienced quite a bit of pain in the past with new guzzle versions and then we'd have to deal with that pain in that repo ...
however now that PSR-7 is around it's a lot less likely that there will be a lot of pain ...

File: .travis.yml [hunk 2]
```diff
@@ -11,6 +11,4 @@ install: composer require --no-update guzzlehttp/guzzle $GUZZLE_VERSION; compose
 script: vendor/bin/phpunit
 
 env:
-  - GUZZLE_VERSION: 4.2
-  - GUZZLE_VERSION: 5.0
-  - GUZZLE_VERSION: 5.2
+  - GUZZLE_VERSION: 6.0
```
Comment (navitronic) [inline .travis.yml]: Guzzle 6 needs php 5.5+, so you will need to remove the earlier `5.4` from this file to have passing tests.

**Comment by navitronic** [comment]
Are you able to update this now that http-signatures has landed PSR-7 support? If not, I can take what you've done here and run with it.

**Comment by rubensayshi** [comment]
yea, updating now! :rocket: 

so essentially the v1.x releases are guzzle 4/5 and this will be tagged as v2.x for guzzle 6 right?  
and incase a bug is ever found that also affects the older v1.x version it needs to be backported right?

**Comment by rubensayshi** [comment]
in the meantime guzzle is already on 6.2, so added a matrix of guzzle versions like it had before

**Comment by rubensayshi** [comment]
ready for review / merge

**Comment by navitronic** [comment]
This looks good.

I am going to merge and test on a project of ours that uses this library.

**Comment by rubensayshi** [comment]
@navitronic time to tag a new version?

**Pull request merged**